netsim/modules/vlan.py

```diff
@@ -296,6 +296,34 @@ def validate_link_vlan_attributes(link: Box,v_attr: Box,topology: Box) -> bool:
                 intf[af] = False                                  # ... make sure it's not connected to the native VLAN subnet
   return link_ok
 
+"""
+check_link_interface_attributes -- validate that the interfaces in bridge/IRB mode do not
+carry any extra attributes
+"""
+def check_link_interface_attributes(link: Box, intf: Box, topology: Box) -> None:
+  valid_attr = list(topology.defaults.vlan.attributes.phy_ifattr) + [ 'node','_vlan_mode' ]
+  vlan_mode = get_vlan_interface_mode(intf)                 # Get interface VLAN forwarding mode
+  if vlan_mode is None or vlan_mode == 'route':             # No access VLAN or routed subinterface
+    return                                                  # ... interface attributes are OK
+
+  invalid = []                                              # This is where we collect invalid attributes
+  for attr in intf.keys():                                  # Now iterate over interface attributes
+    if attr in valid_attr:                                  # L2 attribute below VLANs?
+      continue                                              # ... OK, move on
+    if attr in log.AF_LIST and intf[attr] is False:         # Address family set to False?
```

Comment:
I am working on a proposal for this: Instead of setting ipv4/6: False flags on interfaces, set it in the node vlan(s) and then have the code in ```link_pre_link_transform``` copy them to the interfaces

Reply:
OK, let's see how that turns out. I'm not sure it's worth investing the time though.